File: go/firstPalindrome.go
func firstPalindrome(words []string) string {
    for i:=0; i < len(words); i++{
        if isPal(words[i]){
            return words[i]
        }
    }
    return ""


}

func isPal(s string)bool{
    i, j := 0, len(s)-1

    for i<j && i < len(s) && j >=0 {
        if s[i]!= s[j] {
            return false
        }
        i++
        j--
    }

    return true
}

File: rust/first_palindrome.rs
impl Solution {
    pub fn first_palindrome(words: Vec<String>) -> String {
        for i in 0..words.len(){
            if Self::is_pal(words[i].clone()) {
                return words[i].clone();
            }
        }
        return "".to_string();
    }

    pub fn is_pal(s: String) -> bool {
        let byte_array: &[u8] = s.as_bytes();
        let mut i = 0;
        let mut j = s.len()-1;

        while i < j {
            if byte_array[i]!=byte_array[j]{
                return false;
            }
            i+=1;
            j-=1;
        }

        return true;
    }
}
